Bump org.eclipse.jgit from 4.9.0.201710071750-r to 6.5.0.202303070854-r

NotMyFault opened the pull request:
Like https://github.com/jenkinsci/incrementals-tools/pull/49, but pinning v5 because v6 requires Java 11 or newer and this tool's GH action is bound to Java 8.

Closes https://github.com/jenkinsci/incrementals-tools/pull/49
Closes https://github.com/jenkinsci/incrementals-tools/pull/39

jglick commented:
If it makes anything easier, we could just switch this component to Java 11. I cannot think of any reason why we would need to retain Java 8 compatibility going forward.

NotMyFault commented:
> we could just switch this component to Java 11

Rebased for Java 11. I've added Java 17 to the GH action Java matrix too, given we support Java 17 too. Branch protection rules may need to be adjusted.
